client/src/components/partials/TimeStamp.js

```diff
@@ -0,0 +1,57 @@
+import React from "react";
+
+const TimeStamp = (props) => {
+  const timeStamp = () => {
+    const seconds = Math.floor((new Date() - new Date(props.createdAt).getTime()) / 1000);
+    let interval = Math.floor(seconds / 31536000);
+    let unitsOfTime;
+
+    if (interval >= 1) {
+      unitsOfTime = "year";
+    } else {
+      interval = Math.floor(seconds / 2592000);
+      if (interval >= 1) {
```

Comment:
Hmm at first glance I'm not sure what this does since it's using the same comparator for every if-statements, but I'm sure I'm missing something since you put it there for a reason, so commenting just because I'm curious about how it works

Reply:
The first thing I do is calculate the difference of the time it is now compared to the time a happy thought was created on in milliseconds and divide by 1000 to get that time in seconds. I store that time in a variable called **seconds** on line 5.

Then the variable **interval** which is created on line 6 converts time intervals given in seconds into more human readable form. 

Maybe this breakdown will help you understand it better:
`secondsInAYear = 31536000`
`secondsInAMonth=2592000`
`secondsInADay=86400`
`secondsInAnHour=3600`
`secondsInAMinute=60`

Basically, what is happening is that I take the time the happy thought was created on in seconds and divide it by how many seconds there are in a year, month, day, hour, and minute.

And if the interval is greater than or equal to one I want to add the units of time in string format ("year", "month", "day", "minute", "second") to the end of the actual time that happy thought was created on which is the variable **interval**.

Reply:
I hope that explains how it works in a clear way.